generator/restrictions.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "generator/osm_id.hpp"
+
+#include "std/limits.hpp"
+#include "std/mutex.hpp"
+#include "std/unordered_map.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+class RelationElement;
+
+template <> struct hash<osm::Id>
+{
+  size_t operator()(osm::Id const & id) const
+  {
+    return hash<double>()(id.OsmId());
+  }
+};
+
+/// This class collects all relations with type restriction and save feature ids of
+/// their road feature in text file for using later.
+class RestrictionCollector
+{
+  friend void UnitTest_RestrictionTest_ValidCase();
```

Comment:
Excuse me, I didn't get it, why do we need Inst()? You may create a helper method in RestrictionCollectorTest, named AddFeatureId. Or, you may make m_restrictionCollector protected, and access it directly.